spec/mem/tree_borrows/tree.md

```diff
@@ -4,13 +4,46 @@ The core data structure of Tree Borrows is a *tree*, with a state machine in eac
 We use a tree to track reborrows in each allocation; each reborrow adds a new node to the tree.
 The per-node state machine is defined in [state_machine.md](state_machine.md).
 
-Structurally, we use the usual function representation of a tree: we store a list of children.
+When a reborrow occurs at function entry (i.e. the original reference is passed as an argument), we add a *protector* to the reborrow.
+There are two types of protectors: *strong* and *weak*. A *strong* protector corresponds to a normal reference, while a *weak* protector corresponds to a `Box`.
+The key difference is that only allocations without a strong protector can be deallocated during the execution of a function call.
+We use the following enum to represent whether the node is protected or not, and when it is protected, what type the protector is.
+
+```rust
+enum Protected {
+    Strong,
+    Weak,
+    No,
+}
+
+impl Protected {
+    /// Check whether the node is either strongly or weakly protected.
+    fn yes(self) -> bool {
+        self != Protected::No
+    }
+
+    /// Check whether the node is not protected.
+    fn no(self) -> bool {
+        self == Protected::No
+    }
+
+    /// Check whether the node is actively protected.
+    fn active(self, accessed: Accessed) -> bool {
+        self.yes() && accessed == Accessed::Yes
+    }
+}
+```
+
+Then we can define the node. Structurally, we use the usual functional representation of a tree: we store a list of children.
 
 ```rust
 struct Node {
     children: List<Node>,
     /// State for each location
     location_states: List<LocationState>,
+    /// Indicates whether the node is protected by a function call,
+    /// i.e., whether the original reference passed as an argument of a function call.
```

Comment:
```suggestion
    /// i.e., whether the original reference passed as an argument of a function call.
    /// This will be some kind of protection (weak or strong) if and only if this node is in
    /// some frame's `extra.protectors` list.
```